cmake/PyneMacros.cmake

```diff
@@ -119,6 +119,10 @@ macro(pyne_setup_fortran)
     set(CMAKE_Fortran_FLAGS_RELEASE "-fpic")
     set(CMAKE_Fortran_FLAGS_DEBUG   "-fpic")
   endif(Fortran_COMPILER_NAME MATCHES "gfortran.*")
+
+  # add -fallow-argument-mismatch to fix build with gfortran 10+
+  # https://github.com/pyne/pyne/issues/1416
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch")
```

Comment:
It would be interesting to see whether this fails in our CI configuration without this flag added??  Can you revert this so we can see the failure and then be sure that this resolves it?

Reply:
Good point. I have tested with GNU Fortran (Debian 13.2.0-13) 13.2.0 in my machine. The output was:

```bash
/mnt/Data/Projects/PyNE/src/ensdf_processing/ALPHAD/alphad.f:941:49:

  923 |                CALL CNVU2S(hf(l),dhf(l),hfsav(ihf),LEN(hfsav(ihf)),     &
      |                                 2                
......
  941 |                         CALL CNVU2S(hf(l)-dhf(l),0.0,hfsav(ihf),        &
      |                                                 1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/REAL(4)).
/mnt/Data/Projects/PyNE/src/ensdf_processing/ALPHAD/alphad.f:949:49:

  923 |                CALL CNVU2S(hf(l),dhf(l),hfsav(ihf),LEN(hfsav(ihf)),     &
      |                                 2                
......
  949 |                         CALL CNVU2S(hf(l)+dhf(l),0.0,hfsav(ihf),        &
      |                                                 1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/REAL(4)).
/mnt/Data/Projects/PyNE/src/ensdf_processing/ALPHAD/alphad.f:960:42:

  923 |                CALL CNVU2S(hf(l),dhf(l),hfsav(ihf),LEN(hfsav(ihf)),     &
      |                                 2         
......
  960 |                         CALL CNVU2S(hf(l),0.0,hfsav(ihf),LEN(hfsav(ihf))&
      |                                          1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/REAL(4)).
/mnt/Data/Projects/PyNE/src/ensdf_processing/ALPHAD/alphad.f:1063:30:

  923 |                CALL CNVU2S(hf(l),dhf(l),hfsav(ihf),LEN(hfsav(ihf)),     &
      |                                 2
......
 1063 |             CALL CNVU2S(hf(l),0.0,outstr(5),10,doutst(5),0)
      |                              1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/REAL(4)).
make[2]: *** [src/CMakeFiles/alphad.dir/build.make:75: src/CMakeFiles/alphad.dir/ensdf_processing/ALPHAD/alphad.f.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:248: src/CMakeFiles/alphad.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

Let's see the CI log.

Reply:
Same error with gfortran 11.4 in CI build.